code/script.js

```diff
@@ -218,108 +256,233 @@ const generateBoard = () => {
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
-}
+  audio.pause();
+  winCard.innerHTML = "";
+  totalSeconds = 0;
+  valString = "";
+  secondsLabel.innerHTML = "";
+  minutesLabel.innerHTML = "";
+  attempts = 0;
+  charactersInPlay = CHARACTERS;
+  generateBoard();
+  setSecret();
+  questions.value = "disabled-line";
+};
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const category = questions.options[questions.selectedIndex].parentNode.label;
+  const value = questions.options[questions.selectedIndex].value;
 
   currentQuestion = {
     category: category,
-    // value: value
-  }
-}
+    value: value,
+  };
+};
 
-// This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentQuestion;
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
+  // Refactored version
+  let keep = Array.isArray(secret[category]) ? secret[category].includes(value) : secret[category] === value;
+  // To understnad this refactoring first check the links below:
+  // I have a ternary operator syntax which is a bit confusing but suits for current purpose so u can read about it here:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator
+  // The Array.isArray() method determines whether the passed value is an Array. It returns a Boolean.
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray
+  // In current case if secret[category] is an Array it will continue to this evaluation operation:
+  // secret[category].includes(value)
+  // and return a result of the evaluation in boolean format. In other case if the secret[category]
+  // is not an Array it will return false and continue to
+  // evaluate secret[category] === value , and return a boolean for this evaluation.
 
-  } else if (category === 'accessories' || category === 'other') {
+  //  This code is my initial attempt and is a longer version of 286 line and also works perfectly fine.
+  //   let keep;
+  //   if (category === 'hair' || category === 'eyes') {
+  //     if (secret.hair === value || secret.eyes  === value) {
```

Comment:
It's good that you switched this out for `(secret[category] === value)` because this version with an 'or' actually produces some incorrect filtering if the user happens to choose to check an eye colour of brown and the character has brown hair (but not brown eyes). It would check secret.hair === 'brown' OR secret.eyes === 'brown' and return keep=true when you want it to return false.

Reply:
oh wow , i suppose I just missed this completely. Might be because I swaped the code straight after I wrote it. This bug is actually cause || is returning true if any of conditions is true , right?